### DISCUSSION TITLE
How to limit input (md and css) using glob patterns?

### DISCUSSION THREAD

**Discussion by omundy** [marp-team/marp#328] [Q&A] [answered]
### Limit the input directories using glob?

In my package.json I use: `marp.inputDir: "topics"` to get all `.md` files in this directory. 

I would like to be able to use a glob patterns to get only the first level of markdown files.

For example, this works in the shell: `ls topics/*/*.md` but not in Marp. Is it possible to use glob patterns in inputDir?

### Limit the CSS in themeSet

I set a specific themeSet and location in package.json but Marp is attempting to import some CSS files inside the `topics` directory. How can I prevent this?

```
  "theme": "owen",
  "themeSet": [
      "assets/css/slide-styles.css"
  ],
```

**Comment by yhatt** [comment, marked as answer]
[Marp CLI already supports the glob pattern as Markdown inputs.](https://github.com/marp-team/marp-cli#:~:text=Marp%20CLI%20supports%20converting%20multiple%20files%20by%20passing%20multiple%20paths%2C%20directories%2C%20and%20glob%20patterns.) `topics/*/*.md` should work correctly by avoiding to set `inputDir` option.

```
$ marp "topics/*/*.md"
[  INFO ] Converting 1 markdowns...
[  INFO ] topics/foo/bar.md => topics/foo/bar.html
```

When you did not specify `inputDir` option, Marp CLI will not try to load theme CSS from the directory.

> `inputDir` option is designed mainly for determining the project root directory for the server mode. It tries to load themes from the specified directory automatically. But no worries too much because loaded themes are not used in the slide as long as you have not set the theme in `theme` option / `theme` global directive in Markdown.

**Comment by omundy** [comment]
OK, I see. But if I can't specify an inputDir then how can I change the output directory? 

For example, I have this structure: `/topics/topic-folder/topic.md` 
and I want all the slides to be inside `/slides/topic-folder/topic.html` 
without also pulling in all the md files inside `/topics/topic-folder/topic-example-code/`. 

If I eliminate the inputDir then I can't use output if I want to maintain the same hierarchy for the assets.

https://github.com/omundy/learn-javascript/blob/main/package.json

**Reply by yhatt** [reply]
Try classical `for` loop together with POSIX variable expansion in the shell.

```jsonc
// package.json
{
  "scripts": {
    "build": "for FN in topics/**/*.md; do BFN=${FN%.md}; marp $FN -o slides/${BFN#*/}.html; done"
  }
}
```

```shell
npm run build
```

**Reply by omundy** [reply]
OK. It means I have to wait and then refresh the presentation but it works. Thank you.
